### PR TITLE
Change api of functions Authority.Sign, Authority.Renew

### PR DESCRIPTION
**Pull request by jkralik**
### Description

- Functions Authority.Sign, Authority.Renew returns certificate chain instead of 2 members.
- Extends SignResponse about CertChainPEM.

Implements #126
💔Thank you!

### CODE REVIEW

**Comment by maraino** [comment]
Hi @jkralik, thanks for the contribution, can you please sign the CLA before we can look at this.

**Comment by maraino** [comment]
> Hi @jkralik, thanks for the contribution, can you please sign the CLA before we can look at this.

That was fast

File: api/api.go [hunk 1]
```diff
@@ -211,10 +211,11 @@ func (s *SignRequest) Validate() error {
 
 // SignResponse is the response object of the certificate signature request.
 type SignResponse struct {
-	ServerPEM  Certificate          `json:"crt"`
-	CaPEM      Certificate          `json:"ca"`
-	TLSOptions *tlsutil.TLSOptions  `json:"tlsOptions,omitempty"`
-	TLS        *tls.ConnectionState `json:"-"`
+	ServerPEM    Certificate          `json:"crt"`
+	CaPEM        Certificate          `json:"ca"`
+	CertChainPEM []Certificate        `json:"certChain"`
```
Comment (maraino) [inline api/api.go]: `CaPEM` used to be the issuer certificate (intermediate) and it was used to construct the bundle joining both `ServerPEM` and `CaPEM`, now to construct the bundle we will need to bundle `ServerPEM` with the full `CertChainPEM`. Keeping both should not be necessary, and we should consolidate to one.

File: api/api.go [hunk 2]
```diff
@@ -302,17 +311,22 @@ func (h *caHandler) Sign(w http.ResponseWriter, r *http.Request) {
...
+		caPEM = certChainPEM[1]
+	}
+	logCertificate(w, certChain[0])
 	JSONStatus(w, &SignResponse{
-		ServerPEM:  Certificate{cert},
-		CaPEM:      Certificate{root},
-		TLSOptions: h.Authority.GetTLSOptions(),
+		ServerPEM:    certChainPEM[0],
+		CaPEM:        caPEM,
+		CertChainPEM: certChainPEM,
```
Comment (maraino) [inline api/api.go]: By my previous comment, `CaPEM` or `CertChainPEM` should be `certChainPEM[1:]`

File: api/api.go (hunk 1 above)
Comment (jkralik) [reply api/api.go]: From my point view I want to deprecate/remove `ServerPEM` and `CaPEM` and just leave only `CertChainPEM`. What do you think?

Comment (maraino) [reply api/api.go]: It makes sense to me.

Comment (maraino) [reply api/api.go]: But before making the changes, let me discuss it internally, I know other projects also split the server certificates and intermediates.

Comment (jkralik) [reply api/api.go]: Any news?

Comment (dopey) [reply api/api.go]: Hey @jkralik apologies for the radio silence. We've been going back and forth on how we felt about changing the API (in general we're all for returning the chain). 

This kinda got lost in a jumble between a few other things we've got going on, so we honestly appreciate the reminder. I like the names you've chosen (we argued about names for a while; we like to argue) and I'm giving the PR a last once-over at the moment with the intention of merging / requesting any small changes in a bit.

**Comment by dopey** [comment]
Merged. Thanks for contributing! 🎆

**Pull request merged**